var/spack/repos/builtin/packages/py-mpi4py/package.py

```diff
@@ -31,6 +31,15 @@ class PyMpi4py(PythonPackage):
     def build_args(self, spec, prefix):
         return ['--mpicc=%s -shared' % spec['mpi'].mpicc]
 
+    def setup_build_environment(self, env):
+        # Python is not built with NVHPC, but the compiler flags that were used
+        # to build Python are inherited by the build of py-mpi4py and passed to
+        # NVHPC. This can lead to errors, but by injecting this extra flag we
+        # can demote those errors to warnings.
+        if self.spec.compiler.name == 'nvhpc':
+            env.append_flags('SPACK_CFLAGS', '-noswitcherror')
+            env.append_flags('SPACK_CXXFLAGS', '-noswitcherror')
```

Comment:
Wondering if there is `spack` way to do this instead of changing via env var. @matz-e  might be more up to date on this. (e.g. Is  `self.spec.compiler_flags['cflags']` modifiable ?)

Reply:
@pramodk this is more your area of expertise, I'm just digging around in the Spack guts.

Got to go and see how these things are applied. `spec.compiler_flags` may be a more safe bet, as `SPACK_CFLAGS` may behavior may be switched around depending on how upstream sees fit.

Reply:
I can look a bit more. This:
```python
if self.spec.compiler.name == 'nvhpc':
  for flags in ['cflags', 'cxxflags']:
    self.spec.compiler_flags[flags].append('-noswitcherror')
```
doesn't give an error (i.e. `compiler_flags` is writable), but `-noswitcherror` is not propagated to where it's needed.

Reply:
Ok..we can merge this. I will see later next week if there is anything "better"